teerdays/src/lib.rs

```diff
@@ -0,0 +1,328 @@
+		Bonded { account: T::AccountId, amount: BalanceOf<T> },
+		/// An account's bond has been decreased by an amount
+		Unbonded { account: T::AccountId, amount: BalanceOf<T> },
+		/// An account's accumulated tokentime has been updated
+		TokenTimeUpdated { account: T::AccountId, bond: TeerDayBondOf<T> },
+		/// An account has successfully withdrawn a previously unbonded amount after unlock period has passed
+		Withdrawn { account: T::AccountId, amount: BalanceOf<T> },
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Each account can only bond once
+		AlreadyBonded,
+		/// Insufficient bond
+		InsufficientBond,
+		/// Insufficient unbond
+		InsufficientUnbond,
+		/// account has no bond
+		NoBond,
+		/// Can't unbond while unlock is pending
+		PendingUnlock,
+		/// no unlock is pending
+		NotUnlocking,
+		/// Some corruption in internal state.
+		BadState,
+	}
+
+	/// a store for all active bonds. tokentime is updated lazily
+	#[pallet::storage]
+	#[pallet::getter(fn teerday_bonds)]
+	pub type TeerDayBonds<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::AccountId, TeerDayBondOf<T>, OptionQuery>;
+
+	/// a store for all pending unlocks which are awaiting the unlock period to pass.
+	/// Withdrawal happens lazily and causes entry removal from this store
+	#[pallet::storage]
+	#[pallet::getter(fn pending_unlock)]
+	pub type PendingUnlock<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::AccountId, (T::Moment, BalanceOf<T>), OptionQuery>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight(< T as Config >::WeightInfo::bond())]
+		pub fn bond(
+			origin: OriginFor<T>,
+			#[pallet::compact] value: BalanceOf<T>,
+		) -> DispatchResult {
+			let signer = ensure_signed(origin)?;
+			ensure!(!TeerDayBonds::<T>::contains_key(&signer), Error::<T>::AlreadyBonded);
+			ensure!(value >= T::Currency::minimum_balance(), Error::<T>::InsufficientBond);
+
+			let free_balance = T::Currency::free_balance(&signer);
+			let value = value.min(free_balance);
+			Self::deposit_event(Event::<T>::Bonded { account: signer.clone(), amount: value });
+			T::Currency::set_lock(TEERDAYS_ID, &signer, value, WithdrawReasons::all());
+			let teerday_bond = TeerDayBondOf::<T> {
+				value,
+				last_updated: pallet_timestamp::Pallet::<T>::get(),
+				accumulated_tokentime: BalanceOf::<T>::zero(),
+			};
+			TeerDayBonds::<T>::insert(&signer, teerday_bond);
+			Ok(())
+		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight(< T as Config >::WeightInfo::bond())]
+		pub fn bond_extra(
+			origin: OriginFor<T>,
+			#[pallet::compact] value: BalanceOf<T>,
+		) -> DispatchResult {
+			let signer = ensure_signed(origin)?;
+			ensure!(value >= T::Currency::minimum_balance(), Error::<T>::InsufficientBond);
+			let bond = Self::do_update_teerdays(&signer)?;
+			let free_balance = T::Currency::free_balance(&signer);
+			// free confusingly includes the already bonded amount, so we need to subtract it
```

Comment:
Is this for real? Doesn't that hint that we might make a mistake during the bonding phase?

Reply:
seems to be the same in the staking pallet. very confusing. But given the unit tests, it does what we expect. Will double check how js/apps displays this once we integrate it into the node

Reply:
added a todo here: https://github.com/integritee-network/parachain/issues/298

Reply:
you were right. there is an issue: https://github.com/integritee-network/pallets/pull/265